src/microbe_stage/editor/MicrobeEditor.cs

```diff
@@ -647,22 +647,19 @@ public void SetMembrane(string membraneName)
         gui.UpdateMembraneButtons(Membrane.InternalName);
     }
 
-    public void SetRigidity(float rigidity)
+    public void SetRigidity(int rigidity)
     {
-        if (Math.Abs(Rigidity - rigidity) < MathUtils.EPSILON)
+        if (Math.Abs(Rigidity * 10 - rigidity) < MathUtils.EPSILON * 10)
             return;
 
-        var cost = (int)(Math.Abs(rigidity - Rigidity) / 2 * 100);
+        int cost = (int)(Math.Abs(rigidity - Rigidity * 10) / 2 * 10);
 
         if (cost > 0)
         {
             if (cost > MutationPoints)
-            {
-                rigidity = Rigidity + (rigidity < Rigidity ? -MutationPoints : MutationPoints) * 2 / 100.0f;
-                cost = MutationPoints;
-            }
+                return;
```

Comment:
I think this should change send feedback to the bar to go back to the previous value.